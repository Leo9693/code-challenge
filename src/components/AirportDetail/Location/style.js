import styled, { keyframes } from "styled-components";
import { fadeInRight } from "react-animations";

export const AnimatedLocation = styled.div`
    animation: ${keyframes`${fadeInRight}`} 1.1s;
`;
