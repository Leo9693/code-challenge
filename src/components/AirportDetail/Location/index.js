import React from "react";
import { AnimatedP } from "../style";
import { AnimatedLocation } from "./style";
function Location(props) {
    const {
        aboveSeaLevel,
        latitude,
        latitudeRadius,
        longitude,
        longitudeRadius,
        latitudeDirection,
        longitudeDirection
    } = props.location;
    return (
        <AnimatedLocation>
            <h3>Location:</h3>
            <AnimatedP>aboveSeaLevel: {aboveSeaLevel}</AnimatedP>
            <AnimatedP>latitude: {latitude}</AnimatedP>
            <AnimatedP>latitudeRadius: {latitudeRadius}</AnimatedP>
            <AnimatedP>longitude: {longitude}</AnimatedP>
            <AnimatedP>longitudeRadius: {longitudeRadius}</AnimatedP>
            <AnimatedP>latitudeDirection: {latitudeDirection}</AnimatedP>
            <AnimatedP>longitudeDirection: {longitudeDirection}</AnimatedP>
        </AnimatedLocation>
    );
}

export default Location;
