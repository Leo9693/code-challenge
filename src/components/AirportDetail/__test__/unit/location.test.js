import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Location from "../../Location";
Enzyme.configure({ adapter: new Adapter() });

test("test location Types component", () => {
    const wrapper = shallow(
        <Location
            location={{
                aboveSeaLevel: "",
                latitude: "",
                latitudeRadius: "",
                longitude: "",
                longitudeRadius: "",
                latitudeDirection: "",
                longitudeDirection: ""
            }}
        />
    );
    expect(wrapper.debug()).toMatchSnapshot();
});
