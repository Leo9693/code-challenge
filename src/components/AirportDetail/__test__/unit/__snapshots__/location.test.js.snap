// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test location Types component 1`] = `
"<styled.div>
  <h3>
    Location:
  </h3>
  <styled.p>
    aboveSeaLevel: 
  </styled.p>
  <styled.p>
    latitude: 
  </styled.p>
  <styled.p>
    latitudeRadius: 
  </styled.p>
  <styled.p>
    longitude: 
  </styled.p>
  <styled.p>
    longitudeRadius: 
  </styled.p>
  <styled.p>
    latitudeDirection: 
  </styled.p>
  <styled.p>
    longitudeDirection: 
  </styled.p>
</styled.div>"
`;
